CodeGeneration/Tests/ValidateSyntaxNodes/Validation.swift

```diff
@@ -0,0 +1,136 @@
+        switch choice {
+        case .keyword(text: let keyword):
+          if child.name != "\(keyword.withFirstCharacterUppercased)Keyword" {
+            failures.append(
+              ValidationFailure(
+                node: node.kind,
+                message:
+                  "child '\(child.name)' has a single keyword as its only token choice and should thus be named '\(keyword.withFirstCharacterUppercased)Keyword'"
+              )
+            )
+          }
+        case .token(tokenKind: let tokenKind):
+          switch tokenKind {
+          case "IdentifierToken":
+            // We allow arbitrary naming of identifiers
+            break
+          case "CommaToken":
+            if child.name != "TrailingComma" && child.name != "Comma" {
+              failures.append(
+                ValidationFailure(
+                  node: node.kind,
+                  message: "child '\(child.name)' has a comma keyword as its only token choice and should thus be named 'Comma' or 'TrailingComma'"
+                )
+              )
+            }
+          default:
+            if child.name != tokenKind.dropSuffix("Token") {
+              failures.append(
+                ValidationFailure(
+                  node: node.kind,
+                  message: "child '\(child.name)' has a token as its only token choice and should thus be named '\(tokenKind.dropSuffix("Token"))'"
+                )
+              )
+            }
+          }
+        }
+      }
+    }
+  }
+  return failures
+}
+
+/// If a token only has keyword token choices, its name should end with `Keyword`.
+func validateMultipleKeywordChoicesNaming() -> [ValidationFailure] {
+  var failures: [ValidationFailure] = []
+  for node in SYNTAX_NODES.compactMap(\.layoutNode) {
+    for child in node.children {
+      if case .token(choices: let tokenChoices, _, _) = child.kind,
+         tokenChoices.count > 1, // single token choices are handled by `validateSingleTokenChoiceChildNaming`
+         tokenChoices.allSatisfy({ $0.isKeyword }),
+         !child.name.hasSuffix("Keyword") {
+        failures.append(ValidationFailure(
+          node: node.kind,
+          message: "child '\(child.name)' only has keywords as its token choices and should thus and with 'Keyword'"
+        ))
+      }
+    }
+  }
+  return failures
+}
+
+/// Check that children that have the same type also have the same child name.
+func validateConsistentNamingOfChildren() -> [ValidationFailure] {
+  var failures: [ValidationFailure] = []
+
+  var childrenByNodeKind: [SyntaxNodeKind: [(node: LayoutNode, child: Child)]] = [:]
+
+  for node in SYNTAX_NODES.compactMap(\.layoutNode) {
+    for child in node.children {
+      guard case .node(kind: let childKind) = child.kind else {
+        continue
+      }
+      childrenByNodeKind[childKind, default: []].append((node, child))
+    }
+  }
+
+  for (kind, children) in childrenByNodeKind {
+    if kind.isBase || kind == .token {
+      continue
+    }
```

Comment:
Nit pick: But can this be added to the `for where` IMO it read a but nicer

Reply:
Good idea 👍🏽